### PR TITLE
EventCallback Lib a more modern and easy way to configure

### PR DESCRIPTION
**Pull request by MillhioreBT**
### Pull Request Prelude
- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed
I changed the metatable a bit, now we can access it in a more beautiful way.
These changes allow a special configuration for network events, more information about this special in PR #3838

**Issues addressed:** Nothing!

### CODE REVIEW

**Comment by MillhioreBT** [comment]
I plan to change something soon, so the opinions can wait please, if you have suggestion, here I am

**Review by EPuncker — DISMISSED** [review]

**Comment by DSpeichert** [comment]
> I plan to change something soon, so the opinions can wait please, if you have suggestion, here I am

So is it ready or do you still intend to modify it?

**Comment by MillhioreBT** [comment]
I will remove the byte support, and I will be done, today before I go to sleep I will

**Review by EPuncker — APPROVED** [review]

**Comment by Znote** [comment]
Could you post some updated code samples?
Does this break compatibility with prior versions?

Merge in blocked due to file conflict.

**Review by EPuncker — APPROVED** [review]

**Comment by MillhioreBT** [comment]
> Could you post some updated code samples? Does this break compatibility with prior versions?
> 
> Merge in blocked due to file conflict.

he same, I have simply modified the library to make it more readable, and in theory a little more efficient

**Review by EPuncker — APPROVED** [review]

**Review by DSpeichert — DISMISSED** [review]

**Review by EvilHero90 — DISMISSED** [review]

**Comment by EPuncker** [comment]
there is a conflict

**Review by EPuncker — APPROVED** [review]

File: data/scripts/lib/event_callbacks.lua [hunk 1]
```diff
@@ -1,93 +1,56 @@
+local unpack = unpack
+local pack = table.pack
+
+local EventCallbackData, callbacks, updateableParameters, autoID = {}, {}, {}, 0
+-- This metatable creates an auto-configuration mechanism to create new types of EventCallbacks
+local ec = setmetatable({}, { __newindex = function (self, key, value)
+	autoID = autoID +1
+	callbacks[key] = autoID
+	local info, update = {}, {}
+	for k, v in pairs(value) do if type(k)=="string" then info[k]=v else update[k]=v end end
```
Comment (nekiro) [inline data/scripts/lib/event_callbacks.lua:10]: since when we use dirty coding style?

**Pull request merged**